welcomebot.go

```diff
@@ -10,28 +10,16 @@ import (
 	"github.com/nlopes/slack"
 )
 
-type publicResponse struct {
-	Channel  string `json:"channel"`
-	Raw      bool   `json:"raw_response"`
-	Response string `json:"response"`
-}
-
-type dmResponse struct {
-	Channel  string `json:"channel"`
-	Raw      bool   `json:"raw_response"`
-	Response string `json:"response"`
-}
-
-type ephResponse struct {
+type slackResponse struct {
 	Channel  string `json:"channel"`
 	Raw      bool   `json:"raw_response"`
 	Response string `json:"response"`
 }
 
 type Config struct {
-	PublicResponses []publicResponse `json:"responses"`
-	DmResponses     []dmResponse     `json:"dmresponses"`
-	EphResponses    []ephResponse    `json:"ephresponses"`
+	PublicResponses []slackResponse `json:"responses"`
+	DmResponses     []slackResponse `json:"dmresponses"`
+	EphResponses    []slackResponse `json:"ephresponses"`
 }
```

Comment:
Well spotted. Yeah, no need for these to be separate types if they're all the same. :)